src/components/ChattingScreen.js

```diff
@@ -1,31 +1,125 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import styled from 'styled-components';
 
 import Header from './Header';
 import MessageSender from './MessageSender';
 
 export default function ChattingScreen() {
-	const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
-	const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
-
-	const MSGLIST = [
-		{ user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
-		{ user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
-		{ user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
-		{ user: false, content: '안녕 은아' },
-		{ user: false, content: '뭐해 ?' },
-		{ user: true, content: '시원아 넌 최고의 팀장이야' },
-		{ user: false, content: '나도 아니까  ' },
-		{ user: false, content: '그만 말해줘도 돼' },
-		{ user: true, content: 'ㅠㅠ' },
-	];
-
-	return (
-		<Wrapper>
-			<Header></Header>
-			<MessageSender></MessageSender>
-		</Wrapper>
-	);
+  const EUNKO = 'https://img.techpowerup.org/200908/eun.png';
+  const COOL = 'https://img.techpowerup.org/200908/NjRiY2JjOGU5YzQz.png';
+
+  const MSGLIST = [
+    { user: true, content: '안녕하세요 12기 프론트엔드 개발자분들' },
+    { user: true, content: '저희의 대화를 마음껏 조작해보세요 💌' },
+    { user: true, content: '상단에 프로필을 눌러서 발신자 변경하면 됩니당~' },
+    { user: false, content: '안녕 은아' },
+    { user: false, content: '뭐해 ?' },
+    { user: true, content: '시원아 넌 최고의 팀장이야' },
+    { user: false, content: '나도 아니까  ' },
+    { user: false, content: '그만 말해줘도 돼' },
+    { user: true, content: 'ㅠㅠ' },
+  ];
+
+  const [message, setMessage] = useState(MSGLIST);
+  const [newMessage, setNewMessage] = useState({
+    user: false,
+    content: '',
+  });
+
+  const addNewChat = (e) => {
+    e.preventDefault();
+    if (newMessage.content == '') {
+      alert('type something!!');
+      return;
+    }
+    setMessage([...message, newMessage]);
+    setNewMessage({ ...newMessage, content: '' });
+  };
+
+  console.log(' new', newMessage.content);
+  console.log(' message', ...message);
+
+  const changeInputChat = (e) => {
+    const { name, value } = e.target; //name으로 구별, value는 값 자체
+    setNewMessage({
+      ...newMessage,
+      [name]: value,
+    });
+  };
+
+  const changeUser = (e) => {
+    return newMessage.user
+      ? setNewMessage({ user: false, content: newMessage.content })
+      : setNewMessage({ user: true, content: newMessage.content });
+  };
+
+  useEffect(() => {
+    window.scrollBy(1000, window.innerHeight);
+  }, [newMessage]);
```

Comment:
`useEffect`를 이용할때는, 두번째 인자로 dependency배열을 넘겨줘서 그 인자가 변경되었을때만 호출되도록 적용시킬 수 있는데요,
유빈님의 경우에는 프로필사진을 클릭했을때도 스크롤이 맨 밑으로 내려가게 되어있어요.
프로필사진 클릭으로 `newMessage`안의 `user`의 값이 변해서 상태가 바뀌었기때문에 컴포넌트가 리렌더링되어 `useEffect`안의 함수가 호출된답니당.
그래서 여기서는 `newMessage`가아니라 `message`를 인자로 넘겨줘야지 메시지리스트 전체가 바꼈을때만 호출될 수 있겠죠?
+  message가 전체메시지리스트를 담고있는 변수라는게 직관적이지 않아서 이 변수의 네이밍도 바꿔주시면 좋을 것 같아요 ㅠㅠ

```suggestion
  }, [message]);
```

Reply:
은 완전 똑똑해...

Reply:
헐 생각도 못했네요 감사합니다!!